src/transformState.js

```diff
@@ -5,7 +5,36 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const stateHistory = [];
+
+  for (const order of actions) {
+    const ADD_PROPERTIES = order.type === 'addProperties';
+    const REMOVE_PROPERTIES = order.type === 'removeProperties';
+    const CLEAR_ALL = order.type === 'clear';
+
+    if (ADD_PROPERTIES) {
+      Object.assign(state, order.extraData);
+      stateHistory.push(Object.assign({}, state));
+    }
+
+    if (REMOVE_PROPERTIES) {
+      for (const remove of order.keysToRemove) {
+        delete state[remove];
+      }
+
+      stateHistory.push(Object.assign({}, state));
+    }
+
+    if (CLEAR_ALL) {
+      for (const key in state) {
+        delete state[key];
+      }
+
+      stateHistory.push(Object.assign({}, state));
+    }
+  }
```

Comment:
according to [checklist](https://github.com/mate-academy/js_stateful-object/blob/master/checklist.md) you should use switch statement